src/main/connections/Postgres/PostgresPool.ts

```diff
@@ -0,0 +1,117 @@
+import { Logger } from '@nodescript/logger';
+import { CounterMetric, metric } from '@nodescript/metrics';
+import { config } from 'mesh-config';
+import { dep } from 'mesh-ioc';
+import { Event } from 'nanoevent';
+import pg, { Pool as PoolType } from 'pg';
+
+import { PostgresConnection } from './PostgresConnection.js';
+
+const { Pool } = pg;
+
+
+/**
+ * Encapsulates PostgreSQL Client to facilitate connection pool usage and cleanups.
+ */
+export class PostgresPool {
+
+    @config({ default: 10 }) POOL_SIZE!: number;
+    @config({ default: 10_000 }) CONNECT_TIMEOUT_MS!: number;
+
+    @dep() private logger!: Logger;
+
+
+    becameIdle = new Event<void>();
+
+    pool: PoolType;
+    private createdAt = Date.now();
+    private usedConnections = 0;
+
+    @metric()
+    private connectionStats = new CounterMetric<{
+        type: 'connect' | 'connectionCreated' | 'connectionClosed' | 'close' | 'fail';
+    }>('nodescript_sql_adapter_connections', 'Sql adapter connections');
+
+    constructor(
+        readonly connectionUrl: string,
+        readonly poolKey: string,
+        readonly max: number,
+        readonly connectionTimeoutMillis: number
```

Comment:
These two are named differently in MySqlPool. Imo a shared abstract parent class could be useful here imo to establish an interface and extract common logic.